### PR TITLE
Automate creation of Jenkins credentials

### PR DESCRIPTION
**Pull request by fredg02**
Should those scripts be put in a sub-folder?

### CODE REVIEW

File: jenkins-create-credentials.sh [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+projects_storage_pass_domain="projects-storage.eclipse.org"
+git_eclipse_pass_domain="git.eclipse.org"
+github_pass_domain="github.com"
+gpg_pass_domain="gpg"
+
+
+create_domain_xml() {
+    local domain_name="${1:-}"
+    echo "  Creating domain '${domain_name}'..."
+    ./jenkins-cli.sh "instances/${PROJECT_NAME}" create-credentials-domain-by-xml system::system::jenkins <<EOF
```
Comment (mbarbero) [inline jenkins-create-credentials.sh:42]: $script_folder

File: jenkins-create-credentials.sh [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+}
+
+create_username_password_credentials_xml() {
+    local domain_name="${1:-}"
+    local id="${2:-}"
+    local username="${3:-}"
+    local password="${4:-}"
+    local description="${5:-}"
+    echo "  Creating username/password credential '${id}'..."
+    ./jenkins-cli.sh "instances/${PROJECT_NAME}" create-credentials-by-xml system::system::jenkins "${domain_name}" <<EOF
```
Comment (mbarbero) [inline jenkins-create-credentials.sh:56]: $script_folder

File: jenkins-create-credentials.sh [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+create_ssh_credentials_xml() {
+    local domain_name="${1:-}"
+    local id="${2:-}"
+    local username="${3:-}"
+    local id_rsa="${4:-}"
+    local passphrase="${5:-}"
+    local description="${6:-}"
+    echo "  Creating SSH credential '${id}'..."
+    ./jenkins-cli.sh "instances/${PROJECT_NAME}" create-credentials-by-xml system::system::jenkins "${domain_name}" <<EOF
```
Comment (mbarbero) [inline jenkins-create-credentials.sh:75]: $script_folder

File: jenkins-create-credentials.sh [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    local id="${2:-}"
+    local file_name="${3:-}"
+    local subkeys="${4:-}"
+    echo "  Creating file credential '${id}'..."
+
+    ## TODO: find out how secret bytes need to be encoded
+    echo "  IMPORTANT: file needs to be uploaded manually in the WebUI for now!"
+    #<secretBytes>${subkeys}</secretBytes>
+    
+    ./jenkins-cli.sh "instances/${PROJECT_NAME}" create-credentials-by-xml system::system::jenkins "${domain_name}" <<EOF
```
Comment (mbarbero) [inline jenkins-create-credentials.sh:101]: $script_folder

File: jenkins-create-credentials.sh [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+create_ssh_credentials() {
+    local domain_name="${1:-}"
+    local id="${2:-}"
+    local description="${3:-}"
+    local pass_domain="${4:-}"
+
+    # read credentials from pass
+
+    # exception for projects-storage and git.eclipse.org :/
+    if [[ "${pass_domain}" == "projects-storage.eclipse.org" || "${pass_domain}" == "git.eclipse.org" ]]; then
```
Comment (mbarbero) [inline jenkins-create-credentials.sh:144]: I'm definitely not against adding a username entry to pass for *.eclipse.org entries

**Review by mbarbero — CHANGES_REQUESTED** [review]

File: jenkins-create-credentials-token.sh [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    # read secret from stdin
+    read -p "Secret: " secret
+    if [[ -z "${secret}" ]]; then
+        printf "ERROR: secret must be given.\n"
+        exit 1
+    fi
+
+    # check if credentials already exist
+    reply=$(./jenkins-cli.sh instances/${PROJECT_NAME} get-credentials-as-xml system::system::jenkins ${domain_name} ${id} 2>&1 || true)
```
[the remark this answers is not in the document]
Comment (fredg02) [reply jenkins-create-credentials-token.sh]: Done.

File: jenkins-create-credentials-token.sh [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+if [[ -z "${secret_id}" ]]; then
+ printf "ERROR: a secret id must be given.\n"
+ usage
+ exit 1
+fi
+
+create_domain_xml() {
+    local domain_name="${1:-}"
+    echo "  Creating domain '${domain_name}'..."
+    ./jenkins-cli.sh "instances/${PROJECT_NAME}" create-credentials-domain-by-xml system::system::jenkins <<EOF
```
[the remark this answers is not in the document]
Comment (fredg02) [reply jenkins-create-credentials-token.sh]: Done.

File: jenkins-create-credentials.sh (hunk 1 above)
Comment (fredg02) [reply jenkins-create-credentials.sh]: Done.

File: jenkins-create-credentials.sh (hunk 3 above)
Comment (fredg02) [reply jenkins-create-credentials.sh]: Done.

File: jenkins-create-credentials.sh (hunk 5 above)
Comment (fredg02) [reply jenkins-create-credentials.sh]: Fixed in pass and exception has been removed.

File: jenkins-create-credentials-token.sh [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+EOF
+}
+
+create_string_credentials_xml() {
+    local domain_name="${1:-}"
+    local id="${2:-}"
+    local secret="${3:-}"
+    local description="${4:-}"
+    echo "  Creating string credential '${id}'..."
+    ./jenkins-cli.sh "instances/${PROJECT_NAME}" create-credentials-by-xml system::system::jenkins "${domain_name}" <<EOF
```
[the remark this answers is not in the document]
Comment (fredg02) [reply jenkins-create-credentials-token.sh]: Done.

**Review by mbarbero — APPROVED** [review]

**Pull request merged**